eng/bomgenerator/src/main/java/com/azure/tools/bomgenerator/BomGenerator.java

```diff
@@ -0,0 +1,173 @@
+        List<BomDependency> inputDependencies = new ArrayList<>();
+
+        try {
+            for (String line : Files.readAllLines(Paths.get(inputFileName))) {
+                BomDependency dependency = scanDependency(line);
+
+                if(dependency != null) {
+                    inputDependencies.add(dependency);
+                }
+            }
+        } catch (IOException exception) {
+            exception.printStackTrace();
+        }
+
+        return inputDependencies;
+    }
+
+    private BomDependency scanDependency(String line) {
+        Matcher matcher = SDK_DEPENDENCY_PATTERN.matcher(line);
+        if (!matcher.matches()) {
+            return null;
+        }
+
+        if (matcher.groupCount() != 3) {
+            return null;
+        }
+
+        String artifactId = matcher.group(1);
+        String version = matcher.group(2);
+
+        if(version.contains("-")) {
+            // This is a non-GA library
+            return null;
+        }
+
+        if (EXCLUSION_LIST.contains(artifactId)
+            || artifactId.contains(AZURE_PERF_LIBRARY_IDENTIFIER)
+            || (artifactId.contains(AZURE_TEST_LIBRARY_IDENTIFIER))) {
+            logger.trace("Skipping dependency {}:{}", AZURE_CORE_GROUPID, artifactId);
+            return null;
+        }
+
+        return new BomDependency(AZURE_CORE_GROUPID, artifactId, version);
+    }
+
+    private List<BomDependency> resolveExternalDependencies() {
+        List<BomDependency> externalDependencies = new ArrayList<>();
+        MavenXpp3Reader reader = new MavenXpp3Reader();
+        try {
+            Model model = reader.read(new FileReader(this.pomFileName));
+            DependencyManagement management = model.getDependencyManagement();
+            List<Dependency> externalBomDependencies = management.getDependencies().stream().filter(dependency -> dependency.getType().equals(POM_TYPE)).collect(Collectors.toList());
+            externalDependencies.addAll(Utils.getExternalDependenciesContent(externalBomDependencies));
+        } catch (XmlPullParserException | IOException e) {
+            e.printStackTrace();
+        }
+
+        return externalDependencies;
+    }
+
+    private void rewriteExistingBomFile() {
+        MavenXpp3Reader reader = new MavenXpp3Reader();
+        try {
+            Model model = reader.read(new FileReader(this.pomFileName));
+            DependencyManagement management = model.getDependencyManagement();
+            List<Dependency> dependencies = management.getDependencies();
+            dependencies.sort(new DependencyComparator());
+            management.setDependencies(dependencies);
+            MavenXpp3Writer writer = new MavenXpp3Writer();
+            writer.write(new FileWriter(this.pomFileName), model);
+        } catch (Exception exception) {
+            exception.printStackTrace();
+        }
+    }
+
+    private void writeBom(Collection<BomDependency> bomDependencies) {
```

Comment:
Could this method and the one above be combined? The above only appears to be sorting dependencies while this adds additional dependencies and then sort them. Seems to be duplicative